ncm-ofed/src/main/pan/components/ofed/schema.pan

```diff
@@ -1,115 +1,121 @@
-# ${license-info}
-# ${developer-info}
-# ${author-info}
-
-
-declaration template components/ofed/schema;
+${componentschema}
 
 include 'quattor/types/component';
 
-## openib options (OPENIBOPTS)
+@documentation{openib options}
 type component_ofed_openib_options = {
     "onboot" : boolean = true
 
-    ## MAD datagrams priority
+    @{MAD datagrams priority}
     "renice_ib_mad" : boolean = false
 
-    ## disable for large clusters
+    @{disable CM for IPoIB for large clusters}
     "set_ipoib_cm" : boolean = true
     "set_ipoib_channels" : boolean = false
+    @{IPoIB MTU setting for CM}
+    "ipoib_mtu" : long(0..65536) = 32 * 1024
 
-    ## SRP High Availability
+    @{SRP High Availability}
     "srpha_enable" : boolean = false
     "srp_daemon_enable" : boolean = false
 
-    ## IPoIB MTU setting
-    "ipoib_mtu" : long = 32*1024
-
-    # autotuning
     "run_sysctl" : boolean = true
     "run_affinity_tuner" : boolean = true
+    @{Enable MLNX autotuning}
     "run_mlnx_tune" : boolean = false
 
-    # description
+    @{node description}
     "node_desc" ? string # eg will default to hostname -s
     "node_desc_update_timeout" : long(0..) = 120
     "node_desc_time_before_update" : long(0..) = 10
     "post_start_delay" : long(0..) = 0
 
+    "cx3_eth_only" : boolean = false
 } = dict();
 
-## openib modules (OPENIBMODULES)
+@{openib modules to load}
 type component_ofed_openib_modules = {
     "ucm" : boolean = false
     "umad" : boolean = true
     "uverbs" : boolean = true
 
-    ## RDAM CM (connected mode and unreliable datagram)
+    @{RDMA CM (connected mode) mode}
     "rdma_cm" : boolean = true
+    @{RDMA UD (unreliable datagram) mode}
     "rdma_ucm" : boolean = true
 
-    ## IPoIB
+    @{IPoIB}
     "ipoib" : boolean = true
     "e_ipoib" : boolean = false
 
-    ## SDP (Socket Direct Protocol)
+    @{SDP (Socket Direct Protocol)}
     "sdp" : boolean = false
 
-    ## SRP SCSI RDMA Protocol
+    @{SRP SCSI RDMA Protocol}
     "srp" : boolean = false
-    ## SRP Target
+    @{SRP Target}
     "srpt" : boolean = false
 
-    ## Reliable datagram socket
+    @{Reliable datagram socket}
     "rds" : boolean = false
 
-    ## ISCSI RDMA
+    @{ISCSI RDMA}
     "iser" : boolean = false
 
     "mlx4_vnic" : boolean = false
     "mlx4_fc" : boolean = false
+    "mlx4_en" : boolean = false
 } = dict();
 
-## openib modules (OPENIBHARDWARE)
+@documentation{openib hardware modules to load}
 type component_ofed_openib_hardware = {
-    ## Mellanox
-    ## Inifinihost III
+    @{Mellanox Inifinihost III}
     "mthca" : boolean = false
-    ## Connectx et al
+    @{Mellanox ConnectX}
     "mlx4" : boolean = false
     "mlx5" : boolean = false
 
-    ## Mellanox ethernet
+    @{Mellanox ethernet-only}
     "mlx_en" : boolean = false
 
-    ## Qlogic
+    @{Qlogic}
     "ipath" : boolean = false
     "qib" : boolean = false
```

Comment:
Same here, but I note that `ipath` is deprecated and has been removed (see https://github.com/torvalds/linux/commit/b85d9905a7ca128f24e3a4e60ff2a1b0cd58ae7c)